packages/tsx-editor/src/transpiler/transpile.tsx

```diff
@@ -0,0 +1,37 @@
+import * as monaco from 'monaco-editor';
+import { ITranspiledOutput, IEvalCode } from './transpile.types';
+import { TypeScriptWorker, EmitOutput } from './monacoTypescriptWorker';
+import { ITextModel } from '../components/Editor.types';
+
+export function transpile(model: ITextModel): Promise<ITranspiledOutput> {
+  const ret = new Promise<ITranspiledOutput>(resolve => {
+    monaco.languages.typescript.getTypeScriptWorker().then(_worker => {
+      _worker(model.uri).then((worker: TypeScriptWorker) => {
+        worker.getEmitOutput(model.uri.toString()).then((output: EmitOutput) => {
+          const transpiledOutput: ITranspiledOutput = { error: undefined, outputString: undefined };
```

Comment:
Potentially instead of returning an object with success/error properties, it might be better to just return a rejected promise (by throwing an error) in the error case. The suggested code in my other comment would be modified like this (and update the return type to string):

```ts
  if (output.outputFiles[0]) {
    return output.outputFiles[0].text;
  } else {
    throw new Error('Could not transpile code');
  }
```